### PR TITLE
Add v2 subgraph

### PR DESCRIPTION
**Pull request by FedokDL**

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -152,6 +152,30 @@ type PoolInteraction @entity(immutable: true) {
...
   "The total staked amount after interaction"
   totalStaked: BigInt!
+
+  "The current pool rate"
+  rate: BigInt!
+}
+
+type UserInteraction @entity(immutable: true) {
+  "The tx hash"
+  id: Bytes!
```
Comment (RuslanProgrammer) [inline schema.graphql:162]: We should handle the case when a user make multiple interactions during the same transaction. Maybe use a counter as in the PoolInteraction

File: src/mappings/Distribution.ts [hunk 2]
```diff
@@ -169,15 +204,81 @@ export function handleUserWithdrawn(event: UserWithdrawnEvent): void {
...
+
+  pool.save();
+  userInPool.save();
+}
+
+function _getCurrentPoolRate(address: Address, poolId: BigInt, timestamp: BigInt): BigInt {
+  const distribution = Distribution.bind(address);
+  const poolData = distribution.poolsData(poolId);
+  const rewards = distribution.getPeriodReward(poolId, poolData.getLastUpdate(), timestamp);
+  const rate = poolData.getRate().plus(rewards.div(poolData.getTotalVirtualDeposited()));
```
Comment (RuslanProgrammer) [inline src/mappings/Distribution.ts]: Add check for division by zero?

File: src/mappings/Distribution.ts [hunk 3]
```diff
@@ -169,15 +204,81 @@ export function handleUserWithdrawn(event: UserWithdrawnEvent): void {
   userInPool.staked = userInPool.staked.minus(event.params.amount);
   pool.totalStaked = pool.totalStaked.minus(event.params.amount);
 
+  const poolRate = _getCurrentPoolRate(event.address, event.params.poolId, event.block.timestamp);
+  const userData = _getUserData(event.address, event.params.poolId, event.params.user);
```
Comment (RuslanProgrammer) [inline src/mappings/Distribution.ts:205]: Are these variables not the same?
`poolRate` == `userData.getRate()`?

File: schema.graphql (hunk 1 above)
Comment (FedokDL) [reply schema.graphql:162]: Ok

File: src/mappings/Distribution.ts (hunk 2 above)
Comment (FedokDL) [reply src/mappings/Distribution.ts]: ok

File: src/mappings/Distribution.ts (hunk 3 above)
Comment (FedokDL) [reply src/mappings/Distribution.ts:205]: ok

**Review by RuslanProgrammer — APPROVED** [review]

**Pull request closed without merging**